### PR TITLE
Add storage configuration for edge device

### PR DESCRIPTION
**Pull request by masayag**
The edge-device configuration is extended with stoarge
configuration, to allow edge devices to connect to via
S3 to the storage and upload (or even download ?) from it.

Signed-off-by: Moti Asayag <masayag@redhat.com>

### CODE REVIEW

File: api/v1alpha1/edgedevice_types.go [hunk 1]
```diff
@@ -62,6 +65,14 @@ type HardwareProfileConfiguration struct {
 	Scope string `json:"scope,omitempty"`
 }
 
+type StorageConfiguration struct {
```
Comment (jakub-dzon) [inline api/v1alpha1/edgedevice_types.go:68]: WDYT about nesting the s3-specific configuration to allow for adding other types in the future?
Like:
```yaml
storage:
  s3:
    bucket_host:  ...
    ...
  other_storage_type:
    ...
```

File: api/v1alpha1/edgedevice_types.go [hunk 2]
```diff
@@ -62,6 +65,14 @@ type HardwareProfileConfiguration struct {
...
+	BucketHost         string `json:"bucket_host,omitempty"`
```
Comment (jakub-dzon) [inline api/v1alpha1/edgedevice_types.go:69]: Please use camelCase in json property names for consistency.

File: swagger.yaml [hunk 3]
```diff
@@ -172,6 +174,20 @@ definitions:
           - full
           - delta
 
+  storage-configuration:
```
Comment (jakub-dzon) [inline swagger.yaml:177]: Please nest s3-specific configuration to allow for easy extension with other storage types.

**Review by jakub-dzon — CHANGES_REQUESTED** [review]

File: api/v1alpha1/edgedevice_types.go (hunk 2 above)
Comment (masayag) [reply api/v1alpha1/edgedevice_types.go:69]: there is no consistency in the file :-)

File: api/v1alpha1/edgedevice_types.go (hunk 1 above)
Comment (jakub-dzon) [reply api/v1alpha1/edgedevice_types.go:68]: It might be sftp server, or network drive for example.

**Review by jakub-dzon — APPROVED** [review]

**Review by pkliczewski — APPROVED** [review]

**Pull request merged**